From 232aeaaa5f9e58507ee51d9696ffadd718a43efb Mon Sep 17 00:00:00 2001
From: Christian Berendt <berendt@osism.tech>
Date: Mon, 06 Mar 2023 23:36:57 +0100
Subject: [PATCH] ironic: add enable_ironic_dnsmasq parameter

With the enable_ironic_dnsmasq parameter it is possible
to explicitly disable the ironic-dnsmasq service. By default,
the parameter is set to the value of enable_ironic

Change-Id: Icf94e5d6980ff238eac90e4f8cd8e530477a9c5b
---

diff --git a/ansible/group_vars/all.yml b/ansible/group_vars/all.yml
index 8ddb843..9594eda 100644
--- a/ansible/group_vars/all.yml
+++ b/ansible/group_vars/all.yml
@@ -694,6 +694,7 @@
 enable_horizon_zun: "{{ enable_zun | bool }}"
 enable_influxdb: "{{ enable_cloudkitty | bool and cloudkitty_storage_backend == 'influxdb' }}"
 enable_ironic: "no"
+enable_ironic_dnsmasq: "{{ enable_ironic | bool }}"
 enable_ironic_neutron_agent: "{{ enable_neutron | bool and enable_ironic | bool }}"
 # TODO(yoctozepto): Remove the deprecated enable_ironic_pxe_uefi in Zed.
 enable_ironic_pxe_uefi: "no"
diff --git a/ansible/roles/ironic/defaults/main.yml b/ansible/roles/ironic/defaults/main.yml
index 7684cf9..68875d4 100644
--- a/ansible/roles/ironic/defaults/main.yml
+++ b/ansible/roles/ironic/defaults/main.yml
@@ -77,7 +77,7 @@
   ironic-dnsmasq:
     container_name: ironic_dnsmasq
     group: ironic-inspector
-    enabled: true
+    enabled: "{{ enable_ironic_dnsmasq }}"
     cap_add:
       - NET_ADMIN
     image: "{{ ironic_dnsmasq_image_full }}"
diff --git a/ansible/roles/ironic/tasks/precheck.yml b/ansible/roles/ironic/tasks/precheck.yml
index 7150294..74385d9 100644
--- a/ansible/roles/ironic/tasks/precheck.yml
+++ b/ansible/roles/ironic/tasks/precheck.yml
@@ -72,4 +72,6 @@
       ironic_dnsmasq_dhcp_ranges must be a list
   connection: local
   run_once: True
-  when: not ironic_dnsmasq_dhcp_ranges is sequence
+  when:
+    - enable_ironic_dnsmasq | bool
+    - not ironic_dnsmasq_dhcp_ranges is sequence
diff --git a/releasenotes/notes/enable_ironic_dnsmasq-4288e3e2b5819f2d.yaml b/releasenotes/notes/enable_ironic_dnsmasq-4288e3e2b5819f2d.yaml
new file mode 100644
index 0000000..7429d63
--- /dev/null
+++ b/releasenotes/notes/enable_ironic_dnsmasq-4288e3e2b5819f2d.yaml
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    With the ``enable_ironic_dnsmasq`` parameter it is possible
+    to explicitly disable the ironic-dnsmasq service. By default,
+    the parameter is set to the value of ``enable_ironic``.
